src/Compilers/CSharp/Portable/Binder/RefSafetyAnalysis.cs

```diff
@@ -921,6 +921,24 @@ private void GetAwaitableInstancePlaceholders(ArrayBuilder<(BoundValuePlaceholde
             }
         }
 
+        public override BoundNode? VisitCollectionExpression(BoundCollectionExpression node)
+        {
+            if (node.CollectionCreation is { } collectionCreation)
+            {
+                if (node.CollectionBuilderSpanPlaceholder is { } spanPlaceholder)
+                {
+                    var elementType = ((NamedTypeSymbol)spanPlaceholder.Type!).TypeArgumentsWithAnnotationsNoUseSiteDiagnostics[0];
+                    var safeContext = LocalRewriter.ShouldUseRuntimeHelpersCreateSpan(node, elementType.Type) ? SafeContext.ReturnOnly : _localScopeDepth;
+                    var placeholders = ArrayBuilder<(BoundValuePlaceholderBase, SafeContext)>.GetInstance();
+                    placeholders.Add((spanPlaceholder, safeContext));
+                    using var _ = new PlaceholderRegion(this, placeholders);
```

Comment:
What does this do? Wouldn't the `PlaceholderRegion` get disposed at then end of this block/`if` (ie. before the `Visit` where we need it)?

Reply:
Unfortunately, we cannot (and do not) dispose of `PlaceholderRegion` instances when exiting the scope because ref safety analysis may require traversing sub-expressions multiple times. See [https://github.com/dotnet/roslyn/issues/65961](https://github.com/dotnet/roslyn/issues/65961).

Reply:
Thanks for the additional context. I see that `RemovePlaceholderScope` is commented out and linked to that issue.
As far as the code here, I think it should be:
```
if (...)
{
    ...
    using var _ = new PlaceholderRegion(this, placeholders);
    Visit(collectionCreation);
}
else
{
    Visit(collectionCreation);
}
```

Then the code is correct and clear regardless of the pending bug on `PlaceholderRegion`.